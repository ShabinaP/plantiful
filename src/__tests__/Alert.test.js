import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe('Alert', () => {
    it("renders an error message if action is unsuccessful", () => {
        const { getByText, asFragment } = render(<Alert message="Error!"/>)
        expect(getByText(/error/i)).toHaveTextContent('Error!')
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders a successful message if action is successful", () => {
        const { getByText, asFragment } = render(<Alert message="Success!" success/>)
        expect(getByText(/success/i)).toHaveTextContent('Success!')
        expect(asFragment()).toMatchSnapshot()
    });
    it("if message is empty, it does not render a message", () => {
        const { asFragment } = render(<Alert message="" />)
        expect(asFragment()).toMatchSnapshot()
    })
});