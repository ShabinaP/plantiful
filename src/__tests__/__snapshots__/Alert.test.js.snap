// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert if message is empty, it does not render a message 1`] = `<DocumentFragment />`;

exports[`Alert renders a successful message if action is successful 1`] = `
<DocumentFragment>
  <div
    class="alert alert-true ? 'success' : 'error "
  >
    Success!
  </div>
</DocumentFragment>
`;

exports[`Alert renders an error message if action is unsuccessful 1`] = `
<DocumentFragment>
  <div
    class="alert alert-false ? 'success' : 'error "
  >
    Error!
  </div>
</DocumentFragment>
`;
